### DISCUSSION TITLE
Replace + with custom text

### DISCUSSION THREAD

**Discussion by RajathShanmugaiah** [react-page/react-page#1094] [Q&A] [answered]
I wanted to know if there is anyway to replace the plus sign to a text which adds the new block in the 'autoform'.

As in the current project Im working on requires text than having the + sign in order to give more descriptiveness it would be nice if there is any hack or solution indicated to it.

**Comment by macrozone** [comment]
@RajathShanmugaiah can you be more precise? which plus sign do you mean?

**Reply by RajathShanmugaiah** [reply]
![+](https://user-images.githubusercontent.com/63197365/139081581-36d44021-beab-4a70-928f-aead42d57a8e.PNG)
@macrozone So in the above image I have indicated the plus sign. Currently in order to add another object via the autoform to the array we click on the plus sign, this creates a new set of object with the input fields appended to the array. Can those plus sign be replaced by text to give description?

**Reply by macrozone** [reply]
yeah  that default ListField is a bit ugly and not very nice. I also raised an issue at some point: https://github.com/vazco/uniforms/issues/639

but you can overrride it, let me quickly check

**Comment by macrozone** [comment, marked as answer]
actually there is an easy solution.

Given this schema for example:

```
 controls: {
    type: 'autoform',
    columnCount: 1,
    schema: {
      properties: {
        myArrow: {
          type: 'array',
          uniforms: {
            addIcon: 'add element', // <-- this, you can also use react-nodes here
          },

          items: {
            type: 'object',
            properties: { 
              // ...
            }
          }
       }

```

in this current version this shows a typescript error there, but i will quickly fix that

**Reply by RajathShanmugaiah** [reply]
Solution works perfectly for me!!! Thank you :)

**Reply by RajathShanmugaiah** [reply]
@macrozone  like addIcon does the minus button also called as removeIcon?

**Reply by macrozone** [reply]
did not try, but looking at the code: https://github.com/vazco/uniforms/blob/master/packages/uniforms-material/src/ListField.tsx

```
uniforms: {
  addIcon: "add",
  itemProps: {
     removeIcon: "remove"
  }
}

````

as the (-) is added to every item in the list

**Reply by RajathShanmugaiah** [reply]
The solution is perfect !!! Thank you very much !!! :)
